src/plugins/matrixops/pad.cpp

```diff
@@ -0,0 +1,416 @@
+                                "pad::pad_get_array",
+                                generate_error_message(
+                                    "Unable to create correctly shaped array "
+                                    "from pad width argument"));
+
+                        else
+                        {
+                            blaze::DynamicVector<std::int64_t> v(2, 0);
+                            auto it = w.begin();
+                            //(((before_1, after_1), )), ((((before_1, after_1), ))), ...
+                            // has not been implemented yet
+                            v[0] = extract_scalar_integer_value_strict(*it++);
+                            v[1] = extract_scalar_integer_value_strict(*it);
+                            return primitive_argument_type{std::move(v)};
+                        }
+                    }
+                    else
+                        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                            "pad::pad_get_array",
+                            generate_error_message("syntax error"));
+                }
+            }
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "pad::pad_get_array",
+                generate_error_message("syntax error"));
+        }
+
+        if (args.size() == 2 &&
+            (!is_list_operand_strict(std::move(args[0])) &&
+                !is_list_operand_strict(std::move(args[1]))))
+        {
+            blaze::DynamicVector<std::int64_t> v(2, 0);
+            v[0] = extract_scalar_integer_value_strict(
+                std::move(args[0]), name_, codename_);
+            v[1] = extract_scalar_integer_value_strict(
+                std::move(args[1]), name_, codename_);
+            return primitive_argument_type{std::move(v)};
+        }
+
+        if (args.size() != ndim)
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "pad::pad_get_array",
+                generate_error_message("syntax error"));
+
+        blaze::DynamicMatrix<std::int64_t> m(ndim, 2);
+        std::size_t i = 0;
+        for (auto&& w : args)
+        {
+            if (!is_list_operand_strict(std::move(w)))
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "pad::pad_get_array",
+                    generate_error_message("syntax error"));
+            auto d = extract_list_value_strict(std::move(w), name_, codename_);
+            if (d.size() != 2)
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "pad::pad_get_array",
+                    generate_error_message("syntax error"));
+            auto it = d.begin();
+            if (is_list_operand_strict(*it))
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "pad::pad_get_array",
+                    generate_error_message("syntax error"));
+            m(i, 0) = extract_scalar_integer_value_strict(std::move(*it++));
+            if (is_list_operand_strict(*it))
```

Comment:
Doing a pre-increment here might be (slightly) more efficient that doing the post-increment the line above.